crates/rolldown/tests/rolldown/function/module_types/asset/artifacts.snap

```diff
@@ -4,6 +4,10 @@ snapshot_kind: text
 ---
 # Assets
 
+## assets/foo-vpuLoj-k.txt
+
+BINARY CONTENT DETECTED
```

Comment:
Can we try to display it if it is a valid utf8, the snapshot is misleading.

Reply:
That would lead us to do a pre-check on binary data. I don't against it. But it still doesn't solve the problem of how printing binary files like `.png`.

Or we just print nothing. Any solution suggested?

---

> That would lead us to do a pre-check on binary data.

I thought it again. It a output's content is binary, snapshot the content and compare it between commits is kind of meaningless.

Reply:
FYI, this is esbuild snapshot, I think just print them as ` String::from_utf8_lossy` is good enough rather than `Binary Detected`

Reply:
I changed it to print nothing.

Reply:
For most of the tests the png is small, prints them just ok, I don't see too much garbled char in the esbuild snapshot, they just print the file asset with string https://github.com/evanw/esbuild/blob/d34e79e2a998c21bb71d57b92b0017ca11756912/internal/bundler_tests/snapshots/snapshots_default.txt#L4580-L4626, and with your solution, It is hard to work with esbuild test diff.

Reply:
let's merge it first, if there are too many diff with `file` loader compare to esbuild in the future, I will add the snapshot back.